tests/test_printer.py

```diff
@@ -141,6 +141,15 @@ def test_telemetry(self, printer):
         assert isinstance(item, Telemetry)
         assert item.to_payload() == {'state': 'BUSY'}
 
+    def test_snapshot(self, printer):
+        data = b'1010'
+        printer.snapshot("test", "test", data, time.time())
```

Comment:
How you check what is finger print and if is set right? And what is token and if is set right?